### PR TITLE
operational semantics for senders and schedulers

### PR DESCRIPTION
**Pull request by ericniebler**
Relocates and improves the specification of `submit`'s operational semantics to the `sender` concept. Adds the `scheduler` operational semantics as they were in P1660.

### CODE REVIEW

**Comment by jaredhoberock** [comment]
Looks like there are some conflicts at the moment?

File: wording.md [hunk 1]
```diff
@@ -284,23 +292,26 @@ The name `execution::submit` denotes a customization point object. For some sube
...
 [*Editorial note:* This specification is adapted from `ranges::iter_swap`. *--end editorial note*]
 
 #### `execution::schedule`
 
-The name `execution::schedule` denotes a customization point object. The expression `execution::schedule(S)` for some subexpression `S` is expression-equivalent to:
+The name `execution::schedule` denotes a customization point object. The expression
+`execution::schedule(S)` for some subexpression `S` is expression-equivalent to:
 
-- `S.schedule()`, if that expression is valid and its type `N` models `Sender`. 
+- `S.schedule()`, if that expression is valid and its type `N` models `sender`.
```
Comment (AerialMantis) [inline wording.md]: Should `N` be `S`?

File: wording.md [hunk 2]
```diff
@@ -420,25 +449,68 @@ concept scheduler =
...
 
 - `s` denotes a (possibly const) scheduler object of type `S`,
-- `N` denotes a type that models `Sender`, and
+- `N` denotes a type that models `sender`, and
 - `n` denotes a sender object of type `N`
 
 | Expression | Return Type | Operational semantics |
 |------------|-------------|---------------------- |
-| `execution::schedule(s)` | `N` | Evaluates `execution::schedule(s)` on the calling thread to create `n` |
+| `execution::schedule(s)` | `N` | Evaluates `execution::schedule(s)` on the calling thread to create `N`. \
```
Comment (AerialMantis) [inline wording.md]: I think there might be a typo here - `\ more`?

**Comment by kirkshoop** [comment]
eric is out for the night. The conflicts were created by some of the merges after this was posted. I cannot fix the merges due to other commitments

**Comment by LeeHowes** [comment]
Merged, I think correctly. I may have lost a bit of Eric's formatting in one or two places.

I did not fix the errors Gordon pointed to. I'm not entirely sure if there's a way to create an edit commit in place - my github skills are weak. Jared if you can just patch those in after merging that'd work best.

File: wording.md (hunk 2 above)
Comment (ericniebler) [reply wording.md]: Yes, this is a typo. I can't get to a computer to fix this. You can safely remove the `\ more`.

File: wording.md (hunk 1 above)
Comment (ericniebler) [reply wording.md]: No, but I don't think `N` is filling any purpose here. "... if its type models `sender`" should be fine.

**Comment by jaredhoberock** [comment]
Merged. Thanks for working on this. I'll fix the errors Gordon noticed.

**Pull request merged**